docs/standard-library/openapi.md

```diff
@@ -314,6 +347,94 @@ The openapi3 emitter ignores the "names" for variants in named unions.
 The openapi3 emitter also defines the `@oneOf` decorator which can be specified on a `union` statement to indicate
 that a union should be emitted as a `oneOf` rather than `anyOf`.
 
+## Security Definitions
+
+The OpenAPI emitter uses the takes the [(http) `@useAuth` decorator](https://github.com/microsoft/cadl/blob/main/docs/standard-library/rest/decorators.md#useauth)
+
+#### Examples
+
+The following example shows how to define a security scheme for Azure Active Directory authentication:
+
+```cadl
+@useAuth(AADToken)
+namespace Contoso.WidgetManager;
+@doc("The Azure Active Directory OAuth2 Flow")
+model AADToken
+  is OAuth2Auth<[
+    {
+      type: OAuth2FlowType.authorizationCode;
+      authorizationUrl: "https://api.example.com/oauth2/authorize";
+      tokenUrl: "https://api.example.com/oauth2/token";
+      scopes: ["https://management.azure.com/read", "https://management.azure.com/write"];
+    }
+  ]>;
+```
+
+## Diagnostics
```

Comment:
Would be nice to also have this section auto generated

Reply:
I opened https://github.com/Azure/cadl-azure/issues/2323 to track this issue.